docs/src/main/asciidoc/extension-codestart.adoc

```diff
@@ -0,0 +1,259 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Extension codestart
+include::./attributes.adoc[]
+
+This guide explains how to create and configure a Quarkus Codestart for an extension.
+
+== Description
+
+"Extension Codestarts" is the name we gave to our Quarkus extension quickstart code generation system. It aims to provide a personalized getting started experience with Quarkus.
+A Quarkus extension is able to provide one or more well defined codestarts which will contain the necessary resources and code required to start using that particular extension.
+
+You can play with the extension codestarts in the Quarkus tooling:
+
+* https://code.quarkus.io[code.quarkus.io, window="_blank"] (find the extensions tagged with [code])
+* The Quarkus Maven plugin:
++
+[source,bash]
+----
+mvn io.quarkus:quarkus-maven-plugin:create
+----
+
+* The Quarkus CLI:
++
+[source,bash]
+----
+quarkus create app
+----
+
+== How it works
+
+There are two kinds of codestarts contributing to the generation of a project, the kind where we want to have only one for a project (called Base) and the rest (called Extra).
```

Comment:
I find this sentence a bit confusing but also struggle to suggest something better.

Reply:
I rewrited it